packages/esm-patient-common-lib/src/types/workspace.ts

```diff
@@ -1,14 +1,38 @@
+import { closeWorkspace } from '../workspaces';
+
 /* The possible states a workspace window can be opened in. */
 export type WorkspaceWindowState = 'maximized' | 'hidden' | 'normal';
 
+export interface CloseWorkspaceOptions {
+  /**
+   * Whether to close the workspace ignoring all the changes present in the workspace.
+   *
+   * If ignoreChanges is true, the user will not be prompted to save changes before closing
+   * even if the `testFcn` passed to `promptBeforeClosing` returns `true`.
+   */
+  ignoreChanges?: boolean;
+  /**
+   * If you want to take an action after the workspace is closed, you can pass your function as
+   * `onWorkspaceClose`. This function will be called only after the workspace is closed, given
+   * that the user might be shown a prompt.
+   * @returns void
+   */
+  onWorkspaceClose?: () => void;
+}
+
 /** The default parameters received by all workspaces */
 export interface DefaultWorkspaceProps {
   /**
-   * Call this function to close the workspace. If ignoreChanges is true, the user will not be
-   * prompted to save changes before closing, even if the `testFcn` passed to `promptBeforeClosing`
-   * returns `true`.
+   * Call this function to close the workspace.
+   *
+   * Whenever we close the workspace with `workspace.closeWorkspace()`, the `ignoreChanges`
+   * is set to true by default.
+   *
+   * Whenever you close the opened workspace outside the workspace window, always close the workspace as
+   * `closeWorkspace({ ignoreChanges: false })`, taking care to prompting the user if there
+   * are any pending changes in the workspace.
```

Comment:
If the usually desirable behavior is `ignoreChanges: false`, why isn't that the default?

Reply:
Yes, it's right to have the `ignoreChanges` as false by default. The following are the reasons why it should be `true` by default:

`closeWorkspace` is mainly called at 3 places: The discard button in the form, which should close the workspace directly, the submit button, which also closes the workspace directly, after the endpoint/ process is completed, and the X button at the top. For the X button, I have already passed the `ignoreChanges: false` in the `workspace-window.component.tsx` since it shouldn't directly close the workspace.

Previously, the ignoreChanges was by default set to true, and all the implementations directly call the `closeWorkspace` to close the workspace, which was hence kept the same way in the new implementation. 

Yes, the default behavior should be, not to send the `ignoreChanges`, hence, I am thinking of adding 2 new functions, `discardChangesAndCloseWorkspace` and `closeWorkspaceWithSavedChanges`.

Reply:
I think `ignoreChanges: false` is the preferable default. I think it is fine if we break existing implementations to make it the default. It is likely what the user wants in most cases anyway. It's not going to make anyone's implementation crash or anything. If implementers want to ignore changes they can add the parameter. Interested in @denniskigen and @ibacher 's takes as well.